istioctl/cmd/wait.go

```diff
@@ -255,14 +255,18 @@ func getAndWatchResource(ictx context.Context) *watcher {
 			return err
 		}
 		for w := range watch.ResultChan() {
-			o := w.Object.(metav1.Object)
-			if o.GetName() == nf {
-				result <- strconv.FormatInt(o.GetGeneration(), 10)
-			}
-			select {
-			case <-ictx.Done():
-				return ictx.Err()
-			default:
+			o, okay := w.Object.(metav1.Object)
```

Comment:
Can you explain how !okay can occur? occurs?

Reply:
yes, it will watch resource's change for `istioctl wait`, and then retrieve the generation from resource's annotation. However, there is no annotation function for type of [v1.Status](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Status). In fact, only [v1.Object](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object) should be the validated interface type to wait for the resource's distribution. So just continue should be the correct step.

Reply:
@howardjohn  Sure, for how does !oaky occur? There is a global var named `targetSchema` in this file, and it can be set in function `validateType`. So this !okay will occur when user pass some args which do not implement funciton `GetAnnotations` kind to this command.

Normal:     istioctl experimental wait --for=distribution --threshold=.99 --timeout=20s **virtualservice** bookinfo.default
Abnormal: istioctl experimental wait --for=distribution --threshold=.99 --timeout=20s **status** bookinfo.default